CMakeLists.txt

```diff
@@ -6,15 +6,107 @@ project(mlboard
+# newer, ARMA_64BIT_WORD is enabled by default.
+if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+  # Check the version, to see if ARMA_64BIT_WORD is enabled by default.
+  set(ARMA_HAS_64BIT_WORD 0)
+  if(NOT (${ARMADILLO_VERSION_MAJOR} LESS 5))
+    set(ARMA_HAS_64BIT_WORD 1)
+  else()
+    # Can we open the configuration file?  If not, issue a warning.
+    if(NOT EXISTS "${ARMADILLO_INCLUDE_DIR}/armadillo_bits/config.hpp")
+      message(WARNING "Armadillo configuration file "
+          "(${ARMADILLO_INCLUDE_DIR}/armadillo_bits/config.hpp) does not exist!")
+    else()
+      # We are on a 64-bit system.  Does Armadillo have ARMA_64BIT_WORD enabled?
+      file(READ "${ARMADILLO_INCLUDE_DIR}/armadillo_bits/config.hpp" ARMA_CONFIG)
+      string(REGEX MATCH
+          "[\r\n][ ]*#define ARMA_64BIT_WORD"
+          ARMA_HAS_64BIT_WORD_PRE
+          "${ARMA_CONFIG}")
+
+      string(LENGTH "${ARMA_HAS_64BIT_WORD_PRE}" ARMA_HAS_64BIT_WORD)
+    endif()
+  endif()
+
+  if(ARMA_HAS_64BIT_WORD EQUAL 0)
+    message(WARNING "This is a 64-bit system, but Armadillo was compiled "
+        "without 64-bit index support.  Consider recompiling Armadillo with "
+        "ARMA_64BIT_WORD to enable 64-bit indices (large matrix support). "
+        "mlpack will still work without ARMA_64BIT_WORD defined, but will not "
+        "scale to matrices with more than 4 billion elements.")
+  endif()
+else()
+  # If we are on a 32-bit system, we must manually specify the size of the word
+  # to be 32 bits, since otherwise Armadillo will produce a warning that it is
+  # disabling 64-bit support.
+  if (CMAKE_SIZEOF_VOID_P EQUAL 4)
+    add_definitions(-DARMA_32BIT_WORD)
+  endif ()
+endif()
+
+# Remove this when serialization is removed from mlpack
+set(Boost_ADDITIONAL_VERSIONS
+  "1.49.0" "1.50.0" "1.51.0" "1.52.0" "1.53.0" "1.54.0" "1.55.0")
+find_package(Boost 1.49
+    COMPONENTS
+      serialization
+    REQUIRED
+)
+
+# In Visual Studio, automatic linking is performed, so we don't need to worry
+# about it.  Clear the list of libraries to link against and let Visual Studio
+# handle it.
+if (MSVC)
+  link_directories(${Boost_LIBRARY_DIRS})
+  set(Boost_LIBRARIES "")
+endif ()
+
+link_directories(${Boost_LIBRARY_DIRS})
 
-ADD_SUBDIRECTORY(include/proto)
-INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINARY_DIR})
+if (WIN32)
+  find_package(LAPACK REQUIRED)
+  find_package(BLAS REQUIRED)
+
+  # Piggyback LAPACK and BLAS linking into Armadillo link.
+  set(ARMADILLO_LIBRARIES
+      ${ARMADILLO_LIBRARIES} ${BLAS_LIBRARIES} ${LAPACK_LIBRARIES})
+endif ()
+
+find_package(MLPACK REQUIRED)
+
+find_package(Protobuf REQUIRED)
+
+add_subdirectory(include/proto)
+include_directories(${CMAKE_CURRENT_BINARY_DIR})
+
+set(MLBOARD_INCLUDE_DIRS ${MLPACK_INCLUDE_DIR}
+                        ${ARMADILLO_INCLUDE_DIR}
+                        ${PROTOBUF_INCLUDE_DIR}
+                        )
+
+include_directories(${MMLBOARD_INCLUDE_DIRS})
```

Comment:
There is a double MM over here. Is it intentional?

Reply:
No,  That was a typo